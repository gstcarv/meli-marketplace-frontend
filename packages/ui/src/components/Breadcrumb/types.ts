import { PropsWithChildren } from 'react';

export type BreadcrumbProps = PropsWithChildren;
