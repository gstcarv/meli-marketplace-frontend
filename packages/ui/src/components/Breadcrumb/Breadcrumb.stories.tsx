import type { Meta, StoryObj } from '@storybook/react';
import { Breadcrumb } from './Breadcrumb';
import React from 'react';

const meta = {
    title: 'core/Breadcrumb',
    component: Breadcrumb,
    parameters: {
        layout: 'centered'
    },
    tags: ['autodocs']
} satisfies Meta<typeof Breadcrumb>;

export const Primary: StoryObj<typeof meta> = {
    render: () => (
        <Breadcrumb>
            <Breadcrumb.Item>Home</Breadcrumb.Item>
            <Breadcrumb.Item>Products</Breadcrumb.Item>
            <Breadcrumb.Item>iPhone 15 Pro Max</Breadcrumb.Item>
        </Breadcrumb>
    )
};

export default meta;
