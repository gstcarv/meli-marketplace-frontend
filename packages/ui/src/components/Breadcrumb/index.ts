export { Breadcrumb } from './Breadcrumb';
export type { BreadcrumbProps } from './types';

export { BreadcrumbItem } from './components/BreadcrumbItem';
export type { BreadcrumbItemProps } from './components/BreadcrumbItem/types';
