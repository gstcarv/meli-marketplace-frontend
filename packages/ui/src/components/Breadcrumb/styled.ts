import { breakpoints } from '@meli/design-tokens';
import React, { ComponentProps } from 'react';
import styled from 'styled-components';

export const BreadcrumbContainer = styled.nav`` as React.FC<ComponentProps<'nav'>>;

export const BreadcrumList = styled.ol`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
`;

export const BreadcrumItemSeparator = styled.span`
    font-weight: ${({ theme }) => theme.fontWeight.get('regular')};
    font-size: ${({ theme }) => theme.fontSize.get('caption')};
    color: ${({ theme }) => theme.colors.get('gray-600')};
    margin: 0 ${({ theme }) => theme.spacing.get('sm')};

    @media screen and (max-width: ${breakpoints.get('sm')}) {
        font-size: ${({ theme }) => theme.fontSize.get('caption2')};
    }
`;
