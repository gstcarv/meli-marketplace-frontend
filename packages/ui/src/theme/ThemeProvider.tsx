import 'react-loading-skeleton/dist/skeleton.css';
import { PropsWithChildren } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { GlobalStyle } from '../globalStyles';
import { theme } from './theme';

export const ThemeProvider = (props: PropsWithChildren) => (
    <StyledThemeProvider theme={theme}>
        <>
            {props.children}

            <GlobalStyle />
        </>
    </StyledThemeProvider>
);
