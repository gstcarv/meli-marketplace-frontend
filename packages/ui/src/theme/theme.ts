import { colors, fontSize, radii, spacing, fontWeight } from '../tokens';

export const theme = { colors, fontSize, radii, spacing, fontWeight };

export type Theme = typeof theme;
