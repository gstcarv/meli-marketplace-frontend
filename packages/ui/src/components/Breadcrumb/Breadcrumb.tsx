import { Children, Fragment } from 'react';
import { BreadcrumbItem } from './components/BreadcrumbItem';
import { BreadcrumbContainer, BreadcrumItemSeparator, BreadcrumList } from './styled';
import { BreadcrumbProps } from './types';

export const Breadcrumb = ({ children, ...props }: BreadcrumbProps) => {
    const totalChildren = Children.count(children);

    return (
        <BreadcrumbContainer aria-label='breadcrumb' {...props}>
            <BreadcrumList className='Breadcrumb__list'>
                {Children.map(children, (child, index) => (
                    <Fragment key={index}>
                        {child}{' '}
                        {index !== totalChildren - 1 && (
                            <BreadcrumItemSeparator aria-hidden='true'>|</BreadcrumItemSeparator>
                        )}
                    </Fragment>
                ))}
            </BreadcrumList>
        </BreadcrumbContainer>
    );
};

Breadcrumb.Item = BreadcrumbItem;
