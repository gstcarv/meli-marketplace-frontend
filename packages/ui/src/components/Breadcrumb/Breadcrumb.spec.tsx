import { render } from '@testing-library/react';
import { ThemeProvider } from '../../theme';
import { Breadcrumb } from './Breadcrumb';

describe('<Breadcrumb />', () => {
    it('should render correctly', () => {
        const { container } = render(
            <ThemeProvider>
                <Breadcrumb>
                    <Breadcrumb.Item>item 1</Breadcrumb.Item>

                    <Breadcrumb.Item>item 2</Breadcrumb.Item>

                    <Breadcrumb.Item>item 3</Breadcrumb.Item>

                    <Breadcrumb.Item>item 4</Breadcrumb.Item>
                </Breadcrumb>
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
});
