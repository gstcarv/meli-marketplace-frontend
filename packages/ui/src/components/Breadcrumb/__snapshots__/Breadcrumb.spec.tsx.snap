// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render correctly 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="sc-gtLWhw"
  >
    <ol
      class="sc-egkSDF cdAgkE Breadcrumb__list"
    >
      <li
        class="sc-blHHSb khfedp"
      >
        item 1
      </li>
       
      <span
        aria-hidden="true"
        class="sc-fAUdSK bvBqJh"
      >
        |
      </span>
      <li
        class="sc-blHHSb khfedp"
      >
        item 2
      </li>
       
      <span
        aria-hidden="true"
        class="sc-fAUdSK bvBqJh"
      >
        |
      </span>
      <li
        class="sc-blHHSb khfedp"
      >
        item 3
      </li>
       
      <span
        aria-hidden="true"
        class="sc-fAUdSK bvBqJh"
      >
        |
      </span>
      <li
        class="sc-blHHSb khfedp"
      >
        item 4
      </li>
       
    </ol>
  </nav>
</div>
`;
